File: frontend/src/utils/validators.ts
export const validateEmail = (email: string): boolean => {
  const pattern = /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/
  return pattern.test(email)
}

export const validatePassword = (password: string): boolean => {
  return password.length >= 6
}

export const validatePhone = (phone: string): boolean => {
  const pattern = /^\+?[0-9]{10,15}$/
  return pattern.test(phone)
}

export const validateTaskTitle = (title: string): boolean => {
  return title.trim().length >= 3
}

export const validateHours = (hours: number): boolean => {
  return hours > 0 && hours <= 100
}

export const validateProgress = (progress: number): boolean => {
  return progress >= 0 && progress <= 100
}

export const validateScore = (score: number): boolean => {
  return score >= 0 && score <= 5
}

File: backend/app/utils/validators.py
import re
from typing import Optional

def validate_email(email: str) -> bool:
    """Validate email format"""
    pattern = r'^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$'
    return bool(re.match(pattern, email))

def validate_phone(phone: str) -> bool:
    """Validate phone number format"""
    # Remove common separators
    phone = re.sub(r'[\s\-()]', '', phone)
    pattern = r'^\+?[0-9]{10,15}$'
    return bool(re.match(pattern, phone))

def validate_password(password: str) -> bool:
    """Validate password strength"""
    if len(password) < 6:
        return False
    # At least one number
    if not re.search(r'\d', password):
        return False
    # At least one uppercase letter
    if not re.search(r'[A-Z]', password):
        return False
    return True